### PR TITLE
refactor(simplify inbox state): we are deprecating all the inbox2.0 a…

### PR DESCRIPTION
**Pull request by rileylnapier**
…nd 1.0 UX stuff

we want to get rid of some bloat to make things faster

BREAKING CHANGE: deprecating 2.0 and 1.0 ux

## Description

[Pull Request Description Here]

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-hooks/src/inbox/use-inbox.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ const useInbox = () => {
           return;
         }
 
-        actions.newMessage(courierEvent?.data as ICourierMessage);
+        actions.newMessage(courierEvent?.data as IInboxMessagePreview);
```
Comment (suhasdeshpande) [inline packages/react-hooks/src/inbox/use-inbox.ts:34]: do we need to add any runtime validation for type assertion?

File: packages/react-hooks/src/inbox/use-inbox.ts [hunk 2]
```diff
@@ -64,12 +64,7 @@ const useInbox = () => {
           }
 
           case "unread": {
-            /* 
-              intentionally do not patch unread as if the unread message that is referenced is not in state,
-              then we will be screwing with the local state.  instead just reset last fetched so we fetch new data 
-              the next time the inbox is open and closed
-            */
-            actions.resetLastFetched();
+            actions.markMessageUnread(data.messageId, undefined, true);
```
Comment (suhasdeshpande) [inline packages/react-hooks/src/inbox/use-inbox.ts:67]: `undefined`

Comment (rileylnapier) [inline packages/react-hooks/src/inbox/use-inbox.ts:67]: remove the undefined

File: packages/react-inbox/src/components/Messages2.0/Message.tsx [hunk 3]
```diff
@@ -105,18 +85,16 @@ const MessageContainer = styled.div(({ theme }) => {
...
-  renderTextBlock?: React.FunctionComponent<ITextBlock>;
-  textBlocks: ITextBlock[];
+  preview?: string;
   title?: string;
 }> = ({
   areActionsHovered,
   isMessageHovered,
   read,
   renderedIcon,
-  renderTextBlock,
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Messages2.0/Message.tsx:118]: will need to add a new renderPreview hook

File: packages/react-inbox/src/types.ts [hunk 4]
```diff
@@ -32,15 +26,12 @@ export interface InboxProps {
     markAllAsRead?: string;
     markAsRead?: string;
     markAsUnread?: string;
-    tabs?: string[];
   };
   openLinksInNewTab?: boolean;
   placement?: TippyProps["placement"];
   showUnreadMessageCount?: boolean;
-  tabs?: Array<ITab> | false;
   views?: Array<IView>;
```
Comment (rileylnapier) [inline packages/react-inbox/src/types.ts:33]: remove this for now

File: packages/react-provider/src/transports/types.ts [hunk 5]
```diff
@@ -1,73 +1,50 @@
...
-  data?: ICourierMessage | IInboxMessagePreview | ICourierEventMessage;
+  data?: IInboxMessagePreview | ICourierEventMessage;
 }
 
 export type ICourierEventCallback = (event: ICourierEvent) => void;
 
 export type Interceptor = (
-  message?: ICourierMessage | ICourierEventMessage | IInboxMessagePreview
-) => ICourierMessage | undefined;
+  message?: IInboxMessagePreview | ICourierEventMessage | IInboxMessagePreview
```
Comment (rileylnapier) [inline packages/react-provider/src/transports/types.ts:49]: duplicate type

**Review by suhasdeshpande — DISMISSED** [review]

**Review by suhasdeshpande — APPROVED** [review]

**Pull request merged**